gears/src/types/base/coin/unsigned.rs

```diff
@@ -14,53 +14,64 @@ mod inner {
 
 /// Coin defines a token with a denomination and an amount.
 #[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
-pub struct Coin {
+#[serde(try_from = "inner::Coin", into = "inner::Coin")]
+pub struct UnsignedCoin {
     pub denom: Denom,
     pub amount: Uint256,
 }
 
-impl TryFrom<inner::Coin> for Coin {
-    type Error = CoinsError;
+impl Coin<Uint256> for UnsignedCoin {
```

Comment:
This makes me think that `Coin` amount type should not be a generic argument but an associated type

Reply:
Makes sense

Reply:
Done.